This ensures the xul:scintilla element can receive the focus during a
mouse-click - bug 89902.

Index: /Users/toddw/as/komodo-devel_12/mozilla/build/moz1700-ko8.12/mozilla/content/xul/content/src/nsXULElement.cpp
--- a/content/xul/content/src/nsXULElement.cpp	Thu Oct 18 16:40:12 2012 +1100
+++ b/content/xul/content/src/nsXULElement.cpp	Wed Nov 07 17:05:23 2012 -0800
@@ -521,12 +521,15 @@
 
 // returns true if the element is not a list
 static bool IsNonList(mozilla::dom::NodeInfo* aNodeInfo)
 {
   return !aNodeInfo->Equals(nsGkAtoms::tree) &&
          !aNodeInfo->Equals(nsGkAtoms::listbox) &&
+         // ACTIVESTATE: allow mouse-click focus for xul:scintilla element.
+         !aNodeInfo->NameAtom()->Equals(NS_LITERAL_STRING("scintilla")) &&
+         // ACTIVESTATE: end
          !aNodeInfo->Equals(nsGkAtoms::richlistbox);
 }
 
 bool
 nsXULElement::IsFocusableInternal(int32_t *aTabIndex, bool aWithMouse)
 {
